app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  has_many :comments
+  has_many :likes
+  belongs_to :author, class_name: 'User'
+
+  def update_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
```

Comment:
- [ ] [OPTIONAL] Please use `after_save` to call `update_posts_counter` so this method called for every new post saved to database, then make the `update_posts_counter` method private. it will be like:
```
  after_save :update_posts_counter

  private

  def update_posts_counter
    author.update(posts_counter: author.posts.count)
  end
```

- [ ] [OPTIONAL] Please make the implementation of the `update_posts_counter` like:
```
  def update_posts_counter
    author.increment!(:posts_counter)
  end
```
it's much faster and better for performance. Please make this change for all increase count methods.

Reply:
@gemmen29  Thank you very much for your kind review and feedback. 
- Added